leetcode_bwang

```diff
@@ -0,0 +1,28 @@
+class Solution {
+    public List<List<Integer>> permuteUnique(int[] nums) {
+        List<List<Integer>> wraplist = new ArrayList<>();
+        List<Integer> sublist = new ArrayList<>();
+        boolean[] visited = new boolean[nums.length];
+        Arrays.sort(nums);
+        dfs(wraplist, sublist, nums, visited);
+        return wraplist;
+    }
+    
+    public void dfs(List<List<Integer>> wraplist, List<Integer> sublist, int[] nums, boolean[] visited) {
+        if (sublist.size() == nums.length) {
+            wraplist.add(new ArrayList<>(sublist));
+            return;
+        }
+        
+        for (int i = 0; i < nums.length; i++) {  
+            if (i > 0 && nums[i] == nums[i - 1] && !visited[i - 1]) continue;
+            if (!visited[i]) {
```

Comment:
And then you wanna add number elements into list one by one. Since we are going to add it recursively, we'll need to have some way to keep track of which number has been added in previous calls. There are two ways of doing it: use an index and increment that index in subsequent calls or a boolean array to cache if the element is visited or not.